extensions/omni_schema/src/assemble_schema.sql

```diff
@@ -0,0 +1,271 @@
+create function assemble_schema(conn_info text, fs anyelement, path text default '')
+    returns table
+            (
+                execution_position  int,
+                migration_filename  text,
+                migration_statement text,
+                execution_error     text
+            )
+    language plpgsql
+as
+$$
+declare
+    rec                             record;
+    remote_table_fs                 omni_vfs.table_fs;
+    remote_table_fs_name            text    = 'omni_schema_migration_fs';
+    current_filename                text;
+    current_statement               text;
+    successful_statement_executions int     = 0;
+    error_message                   text;
+    try_again                       boolean = false;
+begin
+    -- statement execution status
+    create temp table omni_schema_execution_status
+    (
+        id                       bigserial primary key,
+        filepath                 text    not null,
+        language                 text,
+        extension                text,
+        file_processor           text,
+        file_processor_extension text,
+        processor                text,
+        processor_extension      text,
+        code                     text    not null,
+        -- order of successful execution
+        execution_number         int,
+        -- whether statement is executed successfully
+        execution_successful     boolean not null default false,
+        -- error for last execution of the statement
+        last_execution_error     text
+    ) on commit drop;
+
+    -- conn for setting up table_fs in remote database
+    perform dblink_connect(remote_table_fs_name, conn_info);
+
+    -- prepare necessary extensions in the remote database
+    for rec in select distinct extension
+               from omni_schema.languages
+               union
+               select distinct file_processor_extension as extension
+               from omni_schema.languages
+               union
+               select distinct processor_extension as extension
+               from omni_schema.auxiliary_tools
+        loop
+            if exists (select from pg_extension where extname = rec.extension) then
```

Comment:
1. `pg_extension` lists installed extensions so shouldn't it be `not exists` ?
2.  couple of issues with creating all language extensions:
  a. not all language extensions will be used by schema files
  b. not all language extensions may be available for installation

Reply:
1. This translates to "if this extension installed locally, then we provision it remotely"
2a. True. Temporary overhead to achieve current goals.
2b. True for truly remote dbs, but arguably not true for the local cluster, where the current use case is.